File: dsa/advent-of-code/2023/gen.py
import os


def create_files(n):
    # Ensure n is formatted as a two-digit string
    n_str = f"{n:02}"

    # Define filenames
    python_file = f"{n_str}.py"
    input_file = f"{n_str}_input.txt"
    sample_file = f"{n_str}_sample.txt"

    # Create and write the Python file
    with open(python_file, "w") as py_file:
        py_file.write(
            f"""
def part_one(data):
    pass

if __name__ == "__main__":
    with open("{n_str}_sample.txt", 'r') as file:
        data = file.readlines()
    print(part_one(data))
"""
        )

    # Create blank text files
    open(input_file, "w").close()
    open(sample_file, "w").close()

    print(f"Files created: {python_file}, {input_file}, {sample_file}")


# Example usage
if __name__ == "__main__":
    n = int(input("Enter a two-digit number (01 to 25): "))
    if 1 <= n <= 25:
        create_files(n)
    else:
        print("Number out of range. Please enter a number between 01 and 25.")

File: dsa/advent-of-code/2017/gen.rb
def create_files(n)
  n_str = n.rjust(2, '0')

  ruby_file = "day_#{n_str}.rb"
  input_file = "day_#{n_str}_input.txt"
  sample_file = "day_#{n_str}_sample.txt"

  File.write(ruby_file, <<~RUBY)
    def part_one(data)
    end

    if __FILE__ == $0
      data = File.readlines("day_#{n_str}_sample.txt")
      part_one(data)
    end
  RUBY

  File.new(input_file, 'w').close
  File.new(sample_file, 'w').close

  puts "Files created: #{ruby_file}, #{input_file}, #{sample_file}"
end

create_files(ARGV[0]) if __FILE__ == $0
